pandas/core/frame.py

```diff
@@ -293,10 +293,17 @@
 
       .. versionadded:: 1.2.0
 
-on : label or list
+on : label or list, or Callable
     Column or index level names to join on. These must be found in both
     DataFrames. If `on` is None and not merging on indexes then this defaults
     to the intersection of the columns in both DataFrames.
+    If Callable this will perform a conditional merge. This allows you to
+    provide a custom match condition for the join. The callable takes 2
+    arguments, which represent the left and right sides of the merge, and
+    must return a boolean. Callable `on` is currently implemented as a chunked
+    cross-join and filtered by the callable provided. Left and right chunk
+    sizes can be set with the `conditional_merge.<left|right>_chunk_size`
```

Comment:
I think it's clearer to list out `get_option("conditional_merge.left_chunk_size")` and `get_option("conditional_merge.right_chunk_size")`, pending my question about equal chunk sizes.

Reply:
no longer relevant, just one option now, see here: https://github.com/pandas-dev/pandas/pull/42964#discussion_r778627377